open-craft/edx-ora2
# SE-2492 rebase and squash ora2 features

This patches the version of edx-ora2 used with juniper.2 (2.7.6) with the changes from the following PRs:

- https://github.com/edx/edx-ora2/pull/1256
- https://github.com/edx/edx-ora2/pull/1250
- https://github.com/edx/edx-ora2/pull/1350

These require the following PRs to be included in the version of edx-platform used:

- https://github.com/edx/edx-platform/pull/23109
- https://github.com/edx/edx-platform/pull/20955

The process used to add the patches was:

1. cherry pick https://github.com/edx/edx-ora2/pull/1256
1. cherry pick https://github.com/edx/edx-ora2/pull/1250
1. cherry pick the first commit from https://github.com/edx/edx-ora2/pull/1350 (the commit with the features excluding the updated translations)
1. regenerate the translations. This had to be done in an extra step because resolving merge conflicts in translations is no fun.
1. bump the version to be something that references 2.7.6 but is different (2.7.6b) so it won't be confused with upstream's version.

**Sandbox URL**: https://se-2492-sandbox.opencraft.hosting/

**Testing instructions**:

See each of the linked pull requests above, and follow the test instructions for each. 

**Author notes and concerns**:

The PR for SE-2010 ( https://github.com/edx/edx-ora2/pull/1350 ) did not apply very cleanly, since it was written for a newer revision of edx-ora2. I believe the main conflicts were related to the recent changes that stripped out python2 support, but we'll see I guess.


**Reviewers**
